apps/api_web/lib/api_web/event_stream.ex

```diff
@@ -7,16 +7,23 @@ defmodule ApiWeb.EventStream do
   import Plug.Conn
   alias __MODULE__.Supervisor
 
-  @type state :: {Plug.Conn.t(), pid}
+  defstruct [:conn, :timer, :timeout, :pid]
```

Comment:
For struct keys that shouldn't be `nil`, like `conn`, `timeout`, and `pid` according to the typespec below, yet nevertheless don't have good defaults, in our apps we use `@enforce_keys`, for an extra level of protection when constructing the struct.

```ex
@enforce_keys [:conn, :timeout, :pid]
defstruct @enforce_keys ++ [:timer]
```

Just an idea, but our team likes it.

Reply:
:+1: added!